src/System.ServiceProcess.ServiceController/tests/ServiceBaseTests.cs

```diff
@@ -164,48 +164,29 @@ public void TestOnContinueBeforePause()
         [ConditionalFact(nameof(IsElevatedAndSupportsEventLogs))]
         public void LogWritten()
         {
-            using (EventLog eventLog = new EventLog("Application"))
-            {
-                ServiceBase sb = new ServiceBase() { ServiceName = nameof(LogWritten) + Guid.NewGuid().ToString() };
-                Assert.False(EventLog.SourceExists(sb.ServiceName));
-                try
-                {
-                    ServiceBase.Run(sb);
-                    eventLog.Source = sb.ServiceName;
-                    Assert.True(EventLog.SourceExists(sb.ServiceName));
-                }
-                finally
-                {
-                    sb.Stop();
-                    EventLog.DeleteEventSource(sb.ServiceName);
-                }
-            }
+            string serviceName = Guid.NewGuid().ToString();
+            // The default username for installing the service is NT AUTHORITY\\LocalService which does not have access to EventLog.
+            // If the username is null, then the service is created under LocalSystem Account which have access to EventLog.
+            var _testService = new TestServiceProvider(serviceName, userName: null);
+            Assert.True(EventLog.SourceExists(serviceName));
+            _testService.DeleteTestServices();
```

Comment:
Why is this called _testService rather than testService?  It's a local.

Reply:
Do we need to do anything else to tear down the testService?

Reply:
Will change to testService

Reply:
No delete test service tears down the service for us